app/src/main/java/ch/epfl/balelecbud/EmergencyInfoActivity.java

```diff
@@ -0,0 +1,52 @@
+package ch.epfl.balelecbud;
+
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.TextView;
+
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.EventListener;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.FirebaseFirestoreException;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+import com.google.firebase.firestore.QuerySnapshot;
+import ch.epfl.balelecbud.Emergency.EmergencyInfo;
+
+public class EmergencyInfoActivity extends BasicActivity {
+
+    private TextView textViewEmergencyInfo;
+    private FirebaseFirestore db = FirebaseFirestore.getInstance();
+    private CollectionReference emergencyInfoRef = db.collection("emergencyInfo");
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_emergency_info);
+        textViewEmergencyInfo = findViewById(R.id.ermergencyTextView);
+    }
+
+    @Override
+    protected void onStart(){
+        super.onStart();
+        emergencyInfoRef.addSnapshotListener(this, new EventListener<QuerySnapshot>() {
+            @Override
+            public void onEvent(QuerySnapshot queryDocumentSnapshots, FirebaseFirestoreException e) {
+                if (e != null) {
+                    return;
+                }
+                String text = "";
+                for (QueryDocumentSnapshot documentSnapshot : queryDocumentSnapshots) {
+                    EmergencyInfo info = documentSnapshot.toObject(EmergencyInfo.class);
+                    text += "Case: "+ info.getName()+"\nWhat to do: "+info.getInstruction()+"\n\n";
```

Comment:
String concatenation in a loop is not recommended because it requires to create new Strings at each iteration, consider using StringBuilder instead.
And try not to hard-code Strings that will be displayed as this prevents easy translation

Reply:
Fixed in https://github.com/karimassi/balelec-bud/pull/31/commits/fe0c3f712418abfc25903063d48847635c381cd1 and https://github.com/karimassi/balelec-bud/pull/31/commits/265d6a295c6c610922f210f19fcc2ba121bbb78d